### PR TITLE
Feature scp244 event transpiler

### PR DESCRIPTION
**Pull request by Undid-Iridium**
Need to do rest of transpilers.

### CODE REVIEW

**Comment by Undid-Iridium** [comment]
FYI, in the future @louis1706 you should not merge your changes to fix non-prefix items into the prefix-branch. Made it very hard for VS2022 to tell what did and did not belong. I had to spent time combing through to verify what I needed

**Comment by Undid-Iridium** [comment]
However, at this point, all 6 transpilers are done.

**Comment by Undid-Iridium** [comment]
I will update this when I am done testing, and done cleaning up.

**Comment by Undid-Iridium** [comment]
Well, bad news. It seems that NW logic bad, probably. It seems that with Dirty Unsafe events that include (Scp244SearchCompletor.Complete), trying to add anything to the "complete" logic is pointless, because it gets removed before the function finishes. I can't really fix that from this end, without making major modifications somewhere, and adding a delay of behavior. Therefore, that one is probably going to remain unfixed. The transpiler works fine, problem is, it can't run. Sorry. I will work on the last two in terms of testing later. For one of your functions, DamageScp244, I changed the logic because it makes more sense for the event to occur at the begging of the function, rather than later on. No point of a isAllowed event if it does not even get called to allow or disallow itself.

**Comment by Undid-Iridium** [comment]
I have tested all 6 transpilers. 4/6 work as intended as far as I can tell, 2/6 do not because of how NW code handles destroying events/functions mid-run because of dirty reads. What do you want to do @louis1706

**Comment by Undid-Iridium** [comment]
Nevermind, this is a skill issue on my side. It seems branch labels did not get transferred correctly, will update when I discover if it is fully me or NW.

**Comment by Undid-Iridium** [comment]
So, I fixed the skill issue; however, there is a problem with how I can call a generic method. For now, the brute force solution to copy your prefix and amend it to what was needed requires deleting code which I did not want to do. I will update with what I can do around that to better handle it (Branching/Deleting, same difference of not wanting).

**Comment by louis1706** [comment]
> So, I fixed the skill issue; however, there is a problem with how I can call a generic method. For now, the brute force solution to copy your prefix and amend it to what was needed requires deleting code which I did not want to do. I will update with what I can do around that to better handle it (Branching/Deleting, same difference of not wanting).

Some part of the code should be deleted for example ``Scp330Interobject::ServerInteract``

			if (!ply.characterClassManager.IsHuman())
			{
				return;
			}
this needs to be deleted

**Comment by Undid-Iridium** [comment]
> > So, I fixed the skill issue; however, there is a problem with how I can call a generic method. For now, the brute force solution to copy your prefix and amend it to what was needed requires deleting code which I did not want to do. I will update with what I can do around that to better handle it (Branching/Deleting, same difference of not wanting).
> 
> Some part of the code should be deleted for example `Scp330Interobject::ServerInteract`
> 
> ```
> 		if (!ply.characterClassManager.IsHuman())
> 		{
> 			return;
> 		}
> ```
> 
> this needs to be deleted
> 
> if you thinks you can by not deleting it talk with nao

I already deleted that code in the, I got InteractingScp330 to work but PickingUpScp244 has a bit of an issue. I will see if it is still a skill issue or NW dirty bit events causing issues

File: Exiled.Events/EventArgs/InteractingScp330EventArgs.cs [hunk 1]
```diff
@@ -31,6 +31,7 @@ public InteractingScp330EventArgs(Player player, CandyKindID candy, int usage, b
             Candy = candy;
             UsageCount = usage;
             ShouldSever = usage >= 2;
+            Log.Info($"What was should sever {ShouldSever}");
```
Comment (louis1706) [inline Exiled.Events/EventArgs/InteractingScp330EventArgs.cs:34]: Should be deleted

**Review by louis1706 — CHANGES_REQUESTED** [review]
(Just for dont miss it)

**Pull request merged**